### PR TITLE
Simplify the cache tests.

### PR DESCRIPTION
**Pull request by mdwn**
Several of the cache tests had a very similar structure. Where possible, these have been adjusted such that all tests with similar structures use a generic testing function that executes the same logic on each resource type.

Note: Let me know if you have any conceptual objection to this. Hopefully it's reasonable as I think it would cut down on a lot of cache boilerplate.

### CODE REVIEW

File: lib/cache/cache_test.go [hunk 1]
```diff
@@ -1268,8 +1275,6 @@ func TestClusterName(t *testing.T) {
 
 // TestNamespaces tests caching of namespaces
 func TestNamespaces(t *testing.T) {
-	t.Parallel()
```
Comment (capnspacehook) [inline lib/cache/cache_test.go:1271]: Why is this removed? What is changed that this test can't be run in parallel anymore?

**Review by capnspacehook — APPROVED** [review]
A good use of generics to clean up code, looks good aside from my one question.

File: lib/cache/cache_test.go (hunk 1 above)
Comment (mdwn) [reply lib/cache/cache_test.go:1271]: Ahhh whoops, that wasn't intentional! I'll re-add that.

**Review by zmb3 — APPROVED** [review]
Lovely diff on this one!

**Review by fspmarshall — APPROVED** [review]
